src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -182,6 +182,11 @@ def exchange(self, algorithm, peer_public_key):
                 _Reasons.UNSUPPORTED_EXCHANGE_ALGORITHM
             )
 
+        if type(peer_public_key.curve) is not type(self.curve):
```

Comment:
Yeah, I wonder if we should just have an `__eq__` on curves.

On Thu, Oct 29, 2015 at 9:37 AM, Ron Frederick <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/ec.py
> <https://github.com/pyca/cryptography/pull/2455#discussion_r43384597>:
>
> > @@ -182,6 +182,11 @@ def exchange(self, algorithm, peer_public_key):
> >                  _Reasons.UNSUPPORTED_EXCHANGE_ALGORITHM
> >              )
> >
> > +        if type(peer_public_key.curve) is not type(self.curve):
>
> I know this was already fixed another way, but for future reference you
> could do it as a type check by saying:
>
>     if not isinstance(peer_public_key.curve, type(self.curve)):
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2455/files#r43384597>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084